engine/src/main/java/com/sandboni/core/engine/sta/operation/DisconnectedTestsOperation.java

```diff
@@ -7,29 +7,38 @@
 import org.jgrapht.Graph;
 
 import java.util.*;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class DisconnectedTestsOperation extends AbstractGraphOperation<SetResult<TestVertex>> {
 
     private final Set<TestVertex> allTests;
     private final Set<TestVertex> relatedTests;
+    private final Set<TestVertex> reflectionCallTests;
     private final Collection<String> sourceLocations;
 
-    DisconnectedTestsOperation(Set<TestVertex> allTests, Set<TestVertex> relatedTests, Collection<String> sourceLocations) {
+    DisconnectedTestsOperation(Set<TestVertex> allTests, Set<TestVertex> relatedTests, Set<TestVertex> reflectionCallTests, Collection<String> sourceLocations) {
         Objects.requireNonNull(allTests, INVALID_ARGUMENT);
         Objects.requireNonNull(relatedTests, INVALID_ARGUMENT);
+        Objects.requireNonNull(reflectionCallTests, INVALID_ARGUMENT);
         Objects.requireNonNull(sourceLocations, INVALID_ARGUMENT);
         this.allTests = new HashSet<>(allTests);
         this.relatedTests = new HashSet<>(relatedTests);
+        this.reflectionCallTests = new HashSet<>(reflectionCallTests);
         this.sourceLocations = new HashSet<>(sourceLocations);
     }
 
     @Override
     public SetResult<TestVertex> execute(Graph<Vertex, Edge> graph) {
         DirectedGraph<Vertex, Edge> directedGraph = (DirectedGraph<Vertex, Edge>) graph;
+
         //getting only the not-ignored and unrelated tests
-        Set<TestVertex> notRelatedTests = allTests.parallelStream()
-                .filter(t -> !t.isIgnore() && !relatedTests.contains(t)).collect(Collectors.toSet());
+        Predicate<TestVertex> nonRelatedPredicate = t -> !t.isIgnore() && !relatedTests.contains(t);
+        Stream<TestVertex> allTestsNonRelatedStream = allTests.parallelStream().filter(nonRelatedPredicate);
+        Stream<TestVertex> reflectionTestsNonRelatedStream = reflectionCallTests.parallelStream().filter(nonRelatedPredicate);
+
+        Set<TestVertex> notRelatedTests = Stream.concat(allTestsNonRelatedStream, reflectionTestsNonRelatedStream).collect(Collectors.toSet());
```

Comment:
Are you sure that reflection tests will be added to disconnected tests here? Seems to me that will be filtered out after the below process to find the disconnected tests. Do you have unit tests for these scenarios?

Reply:
I initially misunderstood the logic for filtering srcLocations.
I've changed the code and added a scenario (see testUnRelatedWithReflectionCall after push)